packages/core/contracts/financial-templates/common/financial-product-libraries/contract-for-difference-libraries/LinearContractForDifferenceFinancialProductLibrary.sol

```diff
@@ -17,35 +31,48 @@ contract LinearContractForDifferenceFinancialProductLibrary is ContractForDiffer
 
     mapping(address => LinearContractForDifferenceParameters) public contractForDifferenceParameters;
 
+    /**
+     * @notice Enables any address to set the parameters price for an associated financial product.
+     * @param contractForDifference address of the CFD contract.
+     * @param upperBound the upper price that the linear CFD will operate within.
+     * @param lowerBound the lower price that the linear CFD will operate within.
+     * @dev Note: a) Any address can set these parameters b) existing CFD parameters for address not set.
+     * c) upperBound > lowerBound.
+     * d) parameters price can only be set once to prevent the deployer from changing the parameters after the fact.
+     * e) For safety, a parameters should be set before depositing any synthetic tokens in a liquidity pool.
+     * f) financialProduct must expose an expirationTimestamp method to validate it is correctly deployed.
+     */
     function setContractForDifferenceParameters(
-        address contractForDifferenceAddress,
+        address contractForDifference,
         int256 upperBound,
         int256 lowerBound
     ) public nonReentrant() {
-        require(
-            ExpiringContractInterface(contractForDifferenceAddress).expirationTimestamp() != 0,
-            "Invalid CFD address"
-        );
+        require(ExpiringContractInterface(contractForDifference).expirationTimestamp() != 0, "Invalid CFD address");
         require(upperBound > lowerBound, "Invalid bounds");
 
-        LinearContractForDifferenceParameters memory params =
-            contractForDifferenceParameters[contractForDifferenceAddress];
+        LinearContractForDifferenceParameters memory params = contractForDifferenceParameters[contractForDifference];
         require(params.upperBound == 0 && params.lowerBound == 0, "Parameters already set");
 
-        contractForDifferenceParameters[contractForDifferenceAddress] = LinearContractForDifferenceParameters({
+        contractForDifferenceParameters[contractForDifference] = LinearContractForDifferenceParameters({
             upperBound: upperBound,
             lowerBound: lowerBound
         });
     }
 
-    function computeExpiraryTokensForCollateral(int256 expiryPrice) public view override returns (uint256) {
+    /**
+     * @notice Returns a number between 0 and 1 to indicate how much collateral each long and short token are entitled
+     * to per collateralPerPair.
+     * @param expiryPrice price from the optimistic oracle for the CFD price identifier.
+     * @return expiryPercentLong to indicate how much collateral should be sent between long and short tokens.
+     */
+    function computeExpiryTokensForCollateral(int256 expiryPrice) public view override returns (uint256) {
         LinearContractForDifferenceParameters memory params = contractForDifferenceParameters[msg.sender];
 
         if (expiryPrice > params.upperBound) return FixedPoint.fromUnscaledUint(1).rawValue;
```

Comment:
I know this isn't a comment on comments, but should this be >= and <= instead of > and <? This way the function could return earlier and save gas for this edge case

Reply:
good point & agreed. have updated